public/app/plugins/datasource/loki/datasource.test.ts

```diff
@@ -1809,6 +1809,192 @@ describe('LokiDatasource', () => {
+
+      await ds.query(query).pipe(take(1)).toPromise();
+
+      expect(fetchMock).toHaveBeenCalledWith(
+        expect.objectContaining({
+          data: expect.objectContaining({
+            queries: expect.arrayContaining([
+              expect.objectContaining({
+                scopes: undefined,
+              }),
+            ]),
+          }),
+        })
+      );
+
+      // Restore original feature toggle values
+      config.featureToggles.scopeFilters = originalScopeFilters;
+      config.featureToggles.logQLScope = originalLogQLScope;
+    });
+
+    it('should handle empty scopes array', async () => {
+      // Enable the required feature toggles
+      const originalScopeFilters = config.featureToggles.scopeFilters;
+      const originalLogQLScope = config.featureToggles.logQLScope;
+      config.featureToggles.scopeFilters = true;
+      config.featureToggles.logQLScope = true;
+
+      const query: DataQueryRequest<LokiQuery> = {
+        ...baseRequestOptions,
+        targets: [{ expr: '{job="grafana"}', refId: 'A' }],
+        scopes: [],
+      };
+
+      const fetchMock = jest.fn().mockReturnValue(of({ data: testLogsResponse }));
+      setBackendSrv({ ...origBackendSrv, fetch: fetchMock });
+
+      await ds.query(query).pipe(take(1)).toPromise();
+
+      expect(fetchMock).toHaveBeenCalledWith(
+        expect.objectContaining({
+          data: expect.objectContaining({
+            queries: expect.arrayContaining([
+              expect.objectContaining({
+                scopes: [],
+              }),
+            ]),
+          }),
+        })
+      );
+
+      // Restore original feature toggle values
+      config.featureToggles.scopeFilters = originalScopeFilters;
+      config.featureToggles.logQLScope = originalLogQLScope;
+    });
+
+    it('should handle undefined scopes', async () => {
+      // Enable the required feature toggles
+      const originalScopeFilters = config.featureToggles.scopeFilters;
+      const originalLogQLScope = config.featureToggles.logQLScope;
+      config.featureToggles.scopeFilters = true;
+      config.featureToggles.logQLScope = true;
+
+      const query: DataQueryRequest<LokiQuery> = {
+        ...baseRequestOptions,
+        targets: [{ expr: '{job="grafana"}', refId: 'A' }],
+        scopes: undefined,
+      };
+
+      const fetchMock = jest.fn().mockReturnValue(of({ data: testLogsResponse }));
+      setBackendSrv({ ...origBackendSrv, fetch: fetchMock });
+
+      await ds.query(query).pipe(take(1)).toPromise();
+
+      expect(fetchMock).toHaveBeenCalledWith(
+        expect.objectContaining({
+          data: expect.objectContaining({
+            queries: expect.arrayContaining([
+              expect.objectContaining({
+                scopes: undefined,
+              }),
+            ]),
+          }),
+        })
+      );
+
+      // Restore original feature toggle values
+      config.featureToggles.scopeFilters = originalScopeFilters;
```

Comment:
Minor, but wouldn't it be slightly better to do this in an afterEach()? I'm personally not a huge fan of doing a lot of stuff outside before/after/test/it/describe.